src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -163,50 +171,38 @@ public void SetDefaultHubName(string hubName)
             this.defaultHubName = hubName;
         }
 
-        internal string GetDebugString()
+        internal void TraceConfiguration(EndToEndTraceHelper traceHelper)
         {
-            var sb = new StringBuilder(4096);
-            sb.AppendLine("Initializing extension with the following settings:");
-            sb.Append(nameof(this.HubName)).Append(":").Append(this.HubName).Append(", ");
-
-            this.AppendStorageProviderValuesToDebugString(sb);
-
-            sb.Append(nameof(this.MaxConcurrentActivityFunctions)).Append(": ").Append(this.MaxConcurrentActivityFunctions).Append(", ");
-            sb.Append(nameof(this.MaxConcurrentOrchestratorFunctions)).Append(": ").Append(this.MaxConcurrentOrchestratorFunctions).Append(", ");
-            sb.Append(nameof(this.ExtendedSessionsEnabled)).Append(": ").Append(this.ExtendedSessionsEnabled).Append(", ");
-            sb.Append(nameof(this.UseGracefulShutdown)).Append(": ").Append(this.UseGracefulShutdown).Append(", ");
-            if (this.ExtendedSessionsEnabled)
-            {
-                sb.Append(nameof(this.ExtendedSessionIdleTimeoutInSeconds)).Append(": ").Append(this.ExtendedSessionIdleTimeoutInSeconds).Append(", ");
-            }
-
-            if (this.NotificationUrl != null)
+            // Clone the options to avoid making changes to the original.
+            // We make updates to the clone rather than to JSON to ensure we're updating what we think we're updating.
+            DurableTaskOptions clone = JObject.FromObject(this).ToObject<DurableTaskOptions>();
+
+            // Don't trace the notification URL query string since it may contain secrets.
+            // This is the only property which we expect to contain secrets. Everything else should be *names*
+            // of secrets that are resolved later from environment variables, etc.
+            if (clone.NotificationUrl != null)
             {
-                // Don't trace the query string, since that contains secrets
-                string url = this.NotificationUrl.GetLeftPart(UriPartial.Path);
-                sb.Append(nameof(this.NotificationUrl)).Append(": ").Append(url).Append(", ");
+                clone.NotificationUrl = new Uri(clone.NotificationUrl.GetLeftPart(UriPartial.Path));
             }
 
-            sb.Append(nameof(this.Notifications)).Append(": { ");
-            this.Notifications.AddToDebugString(sb);
-            sb.Append(" }, ");
+            // At this stage the task hub name is expected to have been resolved. However, we want to know
+            // what the original value was in addition to the resolved value, so we're updating the JSON
+            // blob property to use the original, unresolved value.
+            clone.HubName = this.originalHubName;
```

Comment:
Do we ever log the resolved value now?

Reply:
Yes, all traces include a separate column for the resolved host name.

Reply:
Duh. My bad.